app/Services/NetworkDataService.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Services;
+
+use App\Events\NetworkAircraftDisconnectedEvent;
+use App\Models\Vatsim\NetworkAircraft;
+use Carbon\Carbon;
+use GuzzleHttp\Client;
+use Illuminate\Support\Facades\Log;
+
+class NetworkDataService
+{
+    const NETWORK_DATA_URL = "http://cluster.data.vatsim.net/vatsim-data.json";
+
+    /**
+     * @var Client
+     */
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    public function updateNetworkData(): void
+    {
+        $data = json_decode($this->client->get(self::NETWORK_DATA_URL)->getBody(), true);
+        $this->processClients($data['clients']);
+        $this->handleTimeouts();
+    }
+
+    private function processClients(array $clients): void
+    {
+        foreach ($clients as $client) {
```

Comment:
Would to be more effective to eliminate the controller client types before the iteration?

Reply:
Performance-wise, no - here we do one loop, whereas to eliminate up-front we'd have to do one plus a bit extra :)